packages/base-manager/src/manager-base.ts

```diff
@@ -476,6 +477,42 @@ export abstract class ManagerBase implements IWidgetManager {
     moduleVersion: string
   ): Promise<typeof WidgetModel | typeof WidgetView>;
 
+  protected async loadModelClass(
+    className: string,
+    moduleName: string,
+    moduleVersion: string
+  ): Promise<typeof WidgetModel> {
+    try {
+      const promise: Promise<typeof WidgetModel> = this.loadClass(
+        className,
+        moduleName,
+        moduleVersion
+      ) as Promise<typeof WidgetModel>;
+      await promise;
+      return promise;
+    } catch (error) {
+      return widgets.createErrorWidget(error);
+    }
+  }
+
+  protected async loadViewClass(
+    className: string,
+    moduleName: string,
+    moduleVersion: string
+  ): Promise<typeof WidgetView> {
+    try {
+      const promise: Promise<typeof WidgetView> = this.loadClass(
+        className,
+        moduleName,
+        moduleVersion
+      ) as Promise<typeof WidgetView>;
+      await promise;
+      return promise;
+    } catch (error) {
+      return widgets.createErrorWidgetView(error);
+    }
+  }
```

Comment:
`loadModelClass` and `loadViewClass` are really alike, would it make sense to create only one template function `loadClass<T extends WidgetModel | WidgetView>` (not sure about the TypeScript notation here)?

Reply:
DRY vs KISS, for 2 repeats with low complexity KISS wins :)
Feel free to disagree though.

Reply:
I wouldn't say `loadClass<T>` is more complex, but that's a personal opinion. I don't mind the small repetition.

Reply:
Ah maybe I misunderstood you. The point is that they are different (1 returns a View, the other a Model). I don't see how we can put this in 1 method (like we had before).

Reply:
Ah I missed there was a `loadClass` already, yeah never mind. I should not treat TypeScript like it's C++ :P I've got assumption about the language that might not be correct.